src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/HttpClient/NoOp.cs

```diff
@@ -0,0 +1,55 @@
+
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Api.Experimental;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.SystemExtensions;
+
+namespace NewRelic.Providers.Wrapper.HttpClient
+{
+    public class NoOp : IWrapper
```

Comment:
I think we should spend a little time seeing if we can get away without the need for this wrapper. I'd prefer to avoid needlessly instrumenting things. Ideally, our XML would support the ability to specify a version, but it does not. A step in that direction though would be to see what it would take to check the version of these types in the profiler. If it's easy enough to identify the version at the time of JIT specifically for these methods, then I would be content with special casing these methods in the profiler for now, and putting the XML support in our backlog.

Reply:
The code in the profiler that decides whether to instrument is here https://github.com/newrelic/newrelic-dotnet-agent/blob/e264383d04c924823ed47760647407de4656f0b1/src/Agent/NewRelic/Profiler/MethodRewriter/Instrumentors.h#L30-L34

You may need to pass a handle to `ICorProfilerInfo` into this method so that you can do the appropriate metadata lookup to get the version from either the assembly or module.